### PR TITLE
4125 Immunization programs create UI fixes

### LINKED ISSUE #4125: Immunisation program/vaccine course UI issues
## What went wrong? 😲

https://github.com/msupply-foundation/open-msupply/assets/55115239/df7146ae-12a7-4239-9f12-3163506862ee

- Create modals not clearing on save
- App crashes when try to create duplicate name
- Vaccine course create modal not titled correctly
- Can create duplicate vaccine courses... is that fine?

## How to Reproduce 🔨

Steps to reproduce the behaviour:

See video

### PR DESCRIPTION
**Pull request by lache-melvin**
Fixes #4125

# 👩🏻‍💻 What does this PR do?

Various fixes for Immunization programs/vaccine courses creatiom UI

- handling name already exists error when creating immunisation programs
- clearing inputs on modal close
- fixing the name of the create vaccine course modal

## 💌 Any notes for the reviewer?

# 🧪 Testing

- [x] Input has cleared
  - Go to programs -> Immunizations
  - Click `New Program`
  - Type something in the name
  - Click cancel
  - Click `New Program` again
  - TEST: Input has cleared
- [ ] you see error text under the input that the program already exists
  - Create an Immunization program
  - Try to create another one with the same name
  - TEST: you see error text under the input that the program already exists
- [x] Modal title is now "Add new vaccine course"
  - Click on one of the programs
  - Click `Add new vaccine course`
  - TEST: modal title is now "Add new vaccine course"
- [x] Input has cleared
  - Type something in the name
  - Click cancel
  - Click `Add new vaccine course` again
  - TEST: Input has cleared

# 📃 Documentation

- [x] **Part of an epic**: documentation will be completed for the feature as a whole
- [ ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

File: client/packages/system/src/Immunisation/api/hooks/useImmunisationProgram.ts [hunk 1]
```diff
@@ -126,10 +149,10 @@ const useCreate = () => {
 
 const useUpdate = (
   id: string,
-  setErrorMessage: Dispatch<SetStateAction<string>>
+  setErrorMessage: Dispatch<SetStateAction<string>>,
+  t: TypedTFunction<LocaleKey>
 ) => {
   const { api, storeId, queryClient } = useImmunisationGraphQL();
-  const t = useTranslation('catalogue');
```
Comment (lache-melvin) [inline client/packages/system/src/Immunisation/api/hooks/useImmunisationProgram.ts:132]: Weird one - it was the fact that there is this useTranslation call here that the AppBar was moving down when you clicked opened the create modal... passing in instead 🤷‍♀️

Comment (mark-prins) [reply client/packages/system/src/Immunisation/api/hooks/useImmunisationProgram.ts:132]: I don't like passing around the t function - though for no good reason, just has worked out to be clumsy when I've done it, requiring a lot of parameters. I tested with removing the passed in function and the issue doesn't happen for me now ( I did have it initially and after further testing.. it's all fine :shrug: )

**Review by mark-prins — APPROVED** [review]
👍 LGTM! thanks
one suggestion is to ask if we can have red text for the error here:

Screenshot 2024-06-13 at 10 04 45 AM

and to note that I'm not a fan of passing the t function around.. but it's fine like it is.
Impressive finding that one btw - it's a weird one allright!

File: client/packages/system/src/Immunisation/api/hooks/useImmunisationProgram.ts (hunk 1 above)
Comment (lache-melvin) [reply client/packages/system/src/Immunisation/api/hooks/useImmunisationProgram.ts:132]: The plot thickens! 😅

**Comment by lache-melvin** [comment]
> one suggestion is to ask if we can have red text for the error here

Great, thanks, done here: [d937433](https://github.com/msupply-foundation/open-msupply/pull/4135/commits/d937433bbf2633fb2849f1e8ebe2723545fce74d)

File: client/packages/system/src/Immunisation/ImmunisationProgramDetailView/VaccineCourseCreateModal.tsx [hunk 2]
```diff
@@ -47,7 +47,7 @@ export const VaccineCourseCreateModal: FC<
         />
       }
       cancelButton={<DialogButton variant="cancel" onClick={onClose} />}
-      title={t('label.create-new-program')}
+      title={t('label.add-new-vaccine-course')}
```
[the remark this answers is not in the document]
Comment (lache-melvin) [reply client/packages/system/src/Immunisation/ImmunisationProgramDetailView/VaccineCourseCreateModal.tsx:50]: 🙏 [5432109](https://github.com/msupply-foundation/open-msupply/pull/4135/commits/5432109b09ba0f4db64f25b182afc23c88c7e295)

File: client/packages/system/src/Immunisation/api/hooks/useImmunisationProgram.ts (hunk 1 above)
Comment (lache-melvin) [reply client/packages/system/src/Immunisation/api/hooks/useImmunisationProgram.ts:132]: Hm, I'm still getting it consistently after a page refresh - I agree the passing around is a bit unfortunate... and I don't get it, I swear we conditionally render components with that hook in them all the time??

Will leave for now, maybe future us will have more insight 🤞

**Pull request merged**